service/matching/physical_task_queue_manager.go

```diff
@@ -99,6 +109,9 @@ type (
 		taskValidator              taskValidator
 		tasksAddedInIntervals      *taskTracker
 		tasksDispatchedInIntervals *taskTracker
+		// isDeploymentWorkflowStarted keeps track if we have started a deployment workflow for this
+		// physicalTaskQueue
+		isDeploymentWorkflowStarted atomic.Bool
```

Comment:
This atomic is present to handle the following condition: 
"If a poller arrives with an previously-seen task-queue + buildID combination but with a different DeploymentName (meaning there has already been a Deployment wf started)"

Reply:
> "If a poller arrives with an previously-seen task-queue + buildID combination but with a different DeploymentName (meaning there has already been a Deployment wf started)"

Just for the record, this comment is outdated. a build ID with different deployment name does not have any bearing to other deployments with different deployment names but shared build ID. We care about build ID uniqueness only within a single deployment name.

Reply:
but what if the combination of `task-queue + buildID` is repeated? 

One poller comes in with {TQ: "test", type: Workflow, deployment: "A", buildID: "xyz"} and another poller comes in with  {TQ: "test", type: Workflow, deployment: "B", buildID: "xyz"} 

- Do we want the same combination of task-queue and buildID be present in two different deployments?
- It doesn't break any of the use-cases we have discussed since task-routing will anyways happen by sync updated to UserData and either deployments can make it. I just want to be sure I have the right idea in my head before we proceed

Reply:
In this context, build ID without deployment name is meaningless. So "combination of task-queue + buildID" is meaningless, what is meaningful is combination of task-queue + deployment. And deployment is identified by a composite key: deployment name + build ID.

> Do we want the same combination of task-queue and buildID be present in two different deployments?

So the answer to this is yes, because a TQ can be in multiple deployments, and those deployments may have the same build ID but different deployment names, the same way they may have same deployment names but different build IDs.